### PR TITLE
Integration tests for errors

### PR DESCRIPTION
**Pull request by jtmalinowski**

### CODE REVIEW

File: integration-tests/tests/errors/index.spec.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    browser.globals.clearReceivedSpans();
+    const url = browser.globals.getUrl('/errors/templates/resource-4xx.ejs');
+    await browser.url(url);
+
+    const errorSpan = await browser.globals.findSpan(s => s.name === 'eventListener.error');
+    await browser.assert.ok(!!errorSpan);
+
+    const tags = errorSpan.tags;
+    await browser.assert.ok(tags['location.href'].includes('/errors/templates/resource-4xx.ejs'), `Checking location.href: ${tags['location.href']}`);
+    await browser.assert.strictEqual(tags['splunk.rumVersion'], '0.0.13');
```
Comment (johnbley) [inline integration-tests/tests/errors/index.spec.js:30]: Let's just have each test concentrate on the tags from its data source - location.href, splunk.rumVersion, and app can all be tested by an (e.g.) coretags test or something.

File: integration-tests/tests/errors/index.spec.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    await browser.assert.strictEqual(tags['app'], 'splunk-otel-js-dummy-app');
+    await browser.assert.strictEqual(tags['component'], 'error');
+    await browser.assert.strictEqual(tags['error.type'], 'error');
+    await browser.assert.strictEqual(tags['target_element'], 'IMG');
+    await browser.assert.strictEqual(tags['target_xpath'], '//html/body/img');
+    await browser.assert.ok(tags['target_src'].endsWith('/nonexistent.png'), `Checking target_src: ${tags['target_src']}`);
```
Comment (johnbley) [inline integration-tests/tests/errors/index.spec.js:36]: Nice thought - I've been forgetting to add assert messages to make debugging easier.

File: integration-tests/tests/errors/errors.ejs [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+<head>
+	<meta charset="UTF-8">
+	<title>Errors test</title>
+
+	<%- renderAgent({debug: true}) %>
+</head>
+<body>
+	<h1>FETCH</h1>
+	<script>
+		fetch('/some-data').then(response => response.json()).then(console.log);
```
Comment (johnbley) [inline integration-tests/tests/errors/errors.ejs:12]: Was this just a copy+paste that isn't used any more?

File: integration-tests/tests/errors/index.spec.js [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+    await browser.assert.strictEqual(tags['app'], 'splunk-otel-js-dummy-app');
+    await browser.assert.strictEqual(tags['component'], 'error');
+    await browser.assert.strictEqual(tags['error'], 'true');
+    await browser.assert.strictEqual(tags['error.object'], 'String');
+    await browser.assert.strictEqual(tags['error.message'], 'rejection-value');
+    await browser.assert.strictEqual(tags['ot.status_code'], 'UNSET');
+
+    await browser.end();
+  }
+};
```
Comment (johnbley) [inline integration-tests/tests/errors/index.spec.js:125]: Two tests that I think are missing:
- console.error usage in the app
- usage of SplunkRum.error()

File: integration-tests/tests/errors/index.spec.js [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+      break;
+    }
+
+    await browser.assert.strictEqual(tags['ot.status_code'], 'UNSET');
+
+    await browser.end();
+  },
+  'unhandled promise rejection': async function(browser) {
+    browser.globals.clearReceivedSpans();
+    const url = browser.globals.getUrl('/errors/templates/unhandled-rejection.ejs');
```
Comment (johnbley) [inline integration-tests/tests/errors/index.spec.js:107]: I like that each of these test pages are in separate files; it makes it cleaner.

File: integration-tests/tests/errors/index.spec.js (hunk 1 above)
Comment (jtmalinowski) [reply integration-tests/tests/errors/index.spec.js:30]: makes sense

File: integration-tests/tests/errors/errors.ejs (hunk 3 above)
Comment (jtmalinowski) [reply integration-tests/tests/errors/errors.ejs:12]: Yes 😕

**Comment by jtmalinowski** [comment]
I had to clean the commit history up a little bit, sorry.

**Review by johnbley — APPROVED** [review]
Looking good!

**Review by mhennoch — APPROVED** [review]

File: integration-tests/tests/errors/index.spec.js (hunk 4 above)
Comment (jtmalinowski) [reply integration-tests/tests/errors/index.spec.js:167]: both done

**Pull request merged**